spec/views/request/_sidebar.html.erb_spec.rb

```diff
@@ -1,60 +1,122 @@
-  let(:user) { info_request.user }
-  let(:admin_user) { FactoryBot.create("admin_user") }
-
-  def render_page
-    assign :info_request, info_request
-    assign :track_thing, track_thing
-    assign :status, info_request.calculate_status
-    render
-  end
-
-  context "when the request has been reported" do
-    before :each do
-      info_request.report!("", "", nil)
-    end
-
-    context "and the request is hidden" do
-      it "tell admins it's hidden" do
-        info_request.prominence = "hidden"
-        assign :user, admin_user
-        render_page
-        expect(response).to have_content("This request has prominence " \
-                                         "'hidden'. You can only see it " \
-                                         "because you are logged in as a " \
-                                         "super user.")
-      end
-    end
-
-    context "and the request is requester only" do
-      it "should tell the user that only they can see it" do
-        info_request.prominence = "requester_only"
-        assign :user, user
-        render_page
-        expect(response).to have_content("This request is hidden, so that " \
-                                         "only you the requester can see " \
-                                         "it. Please contact us if you are " \
-                                         "not sure why.")
-      end
-    end
-
-    context "and then deemed okay and left to complete" do
-      it "should let the user know that the admins have not hidden it" do
-        info_request.set_described_state("successful")
-        render_page
-        expect(response).to have_content("This request has been marked for " \
-                                         "review by the site " \
-                                         "administrators, who have not " \
-                                         "hidden it at this time.")
-      end
-    end
-  end
-end
\ No newline at end of file
+require 'spec_helper'
+
+describe 'request/sidebar' do
+  def render_view
+    render partial: self.class.top_level_description,
+           locals: stub_locals
+  end
+
+  let(:info_request) { FactoryBot.build(:info_request) }
+  let(:similar_requests) { double.as_null_object }
+  let(:similar_more) { double.as_null_object }
+
+  let(:stub_locals) do
+    { info_request: info_request,
+      similar_requests: similar_requests,
+      similar_more: similar_more }
+  end
+
+  it 'renders the new request CTA' do
+    render_view
+    expect(rendered).to render_template(partial: 'general/_new_request')
+  end
+
+  it 'renders the pro upsell' do
+    render_view
+    expect(rendered).to render_template(partial: '_sidebar_pro_upsell')
+  end
+
+  context 'when the user can create_embargo', feature: :alaveteli_pro do
+    before do
+      ability = Object.new.extend(CanCan::Ability)
+      ability.can :create_embargo, info_request
+      allow(controller).to receive(:current_ability).and_return(ability)
```

Comment:
This stubing of an ability would be good to extract out into a helper at some point